src/generators/DetailsPageGenerator.ts

```diff
@@ -39,15 +49,37 @@ export const createDetailPage = (createPage: any) => {
   }
 };
 
-const createNutritionTable = ({ createPageFunction, foods, indexFileName }: any) => {
+const createNutritionTable = ({ createPageFunction, foods }: any) => {
   let ffSearchIndex: any = [];
   const template = path.resolve("src/templates/DetailsPage.tsx");
-  foods.forEach((food: any) => {
+  foods.forEach((food: Food) => {
     if (food.name) {
+      function filterNutrient(food: Food, name: string, unit: string): number {
```

Comment:
You are creating functions (`filterNutrient`, `generateSEOInfo`) for each food. This is not correct. Create them once, use them multiple times by passing data.

Reply:
Fixed